### PR TITLE
publish the sub tests result in openQA

### PR DESCRIPTION
**Pull request by Julie-CAO**
my previous push is not complete. It results in the mass failure of the rest runs. The push is to print out the test output to the openQA console then parse the test result. 

- The Trello card: https://trello.com/c/tOdrk5oq/212-p0-openqaguestupgradelog-parser-couldnt-present-the-sub-test-result
- Verification run: http://10.67.18.247/tests/100

@alice-suse @XGWang0 @xguo @waynechen55

### CODE REVIEW

**Comment by alice-suse** [comment]
This PR is necessary for GMC milestone test. So need to handle in high priority.

Code LGTM. Please do not forget to merge all commits into one.

**Comment by alice-suse** [comment]
@mitiao Would you please also help to review it? Hope to be merged ASAP after quick fix if any.

File: tests/virt_autotest/guest_upgrade_run.pm [hunk 1]
```diff
@@ -49,19 +49,43 @@ sub analyzeResult {
             $result->{$testcase_name}->{error}  = $error             ? $error   : 'none';
         }
     }
+
+    #if there are failing guest installation
```
Comment (mitiao) [inline tests/virt_autotest/guest_upgrade_run.pm]: Please use # followed by a single space comment style as we do.

File: tests/virt_autotest/guest_upgrade_run.pm [hunk 2]
```diff
@@ -49,19 +49,43 @@ sub analyzeResult {
...
     repl_module_in_sourcefile();
+
+    $self->execute_script_run("[ -d /var/log/qa/ctcs2 ] && rm -r /var/log/qa/ctcs2 ; [ -d /tmp/prj4_guest_upgrade ] && rm -r /tmp/prj4_guest_upgrade", 30);
+
+    $self->run_test($timeout, "", "no", "yes", "/var/log/qa/", $upload_log_name);
+
+    #display test result
+    # print the test output to the openQA output
+    my $cmd                       = "cd /tmp; zcat $upload_log_name.tar.gz | sed -n '/Executing VM installation/,/[0-9]* fail [0-9]* succeed/p'";
+    my $guest_upgrade_log_content = &script_output("$cmd");
```
Comment (mitiao) [inline tests/virt_autotest/guest_upgrade_run.pm]: IMO, in the most circumstances to call a sub without using `&`

**Comment by mitiao** [comment]
Please rebase commits to one and have upper letter in the beginning

File: tests/virt_autotest/guest_upgrade_run.pm [hunk 3]
```diff
@@ -49,19 +49,43 @@ sub analyzeResult {
...
+    }
     return $result;
 }
 
 sub run {
-    my $self = shift;
-    my $timeout = get_var("MAX_TEST_TIME", "36000") + 10;
+    my $self            = shift;
+    my $timeout         = get_var("MAX_TEST_TIME", "36000") + 10;
+    my $upload_log_name = "guest-upgrade-logs";
```
Comment (mitiao) [inline tests/virt_autotest/guest_upgrade_run.pm]: Better use `''` to quote plain texts.

File: tests/virt_autotest/guest_upgrade_run.pm (hunk 1 above)
Comment (Julie-CAO) [reply tests/virt_autotest/guest_upgrade_run.pm]: no problem

**Comment by Julie-CAO** [comment]
rebased and waiting for the tidy check.

**Comment by Julie-CAO** [comment]
Tidy check passed. Can someone help with merging into the master? thanks.

**Pull request merged**